ref_arch/multi_tenant.rst

```diff
@@ -0,0 +1,540 @@
+    FOREIGN KEY (company_id, ad_id)     -- added
+      REFERENCES ads (company_id, id)
+  );
+
+Distributing Tables, Ingesting Data
+-----------------------------------
+
+.. note::
+
+  This guide is designed so you can follow along in your own Citus database. Use one of these alternatives to spin up a database:
+
+  * Run Citus locally using :ref:`single_machine_docker`, or
+  * Provision a cluster using `Citus Cloud <https://console.citusdata.com/users/sign_up>`_
+
+  You'll run the SQL commands using psql:
+
+  * **Docker**: :code:`docker exec -it citus_master psql -U postgres`
+  * **Cloud**: :code:`psql "connection-string"` where the connection string for your formation is available in the Cloud Console.
+
+  In either case psql will be connected to the coordinator node for the cluster.
+
+At this point feel free to follow along in your own Citus cluster by `downloading <https://examples.citusdata.com/tutorial/schema.sql>`_ and executing the SQL to create the schema. Once the schema is ready, we can tell Citus to create shards on the workers. From the coordinator node, run:
+
+::
+
+  SELECT create_distributed_table('companies',   'id');
+  SELECT create_distributed_table('campaigns',   'company_id');
+  SELECT create_distributed_table('ads',         'company_id');
+  SELECT create_distributed_table('clicks',      'company_id');
+  SELECT create_distributed_table('impressions', 'company_id');
+
+The :ref:`create_distributed_table` function informs Citus that a table should be distributed and that future incoming queries to those tables should be planned for distributed execution. The function also creates shards for the table on worker nodes, allowing queries to be routed there.
+
+The next step is loading sample data into the cluster.
+
+.. code-block:: bash
+
+  # download and ingest datasets from the shell
+
+  for dataset in companies campaigns ads clicks impressions; do
+    curl -O https://examples.citusdata.com/tutorial/${dataset}.csv
+  done
+
+.. note::
+
+  **If you are using Docker,** you should use the :code:`docker cp` command to copy the files into the Docker container.
+
+  .. code-block:: bash
+
+    docker cp companies.csv citus_master:.
+    docker cp campaigns.csv citus_master:.
+    docker cp ads.csv citus_master:.
+
+Being an extension of PostgreSQL, Citus supports bulk loading with the COPY command. Use it to ingest the data you downloaded, and make sure that you specify the correct file path if you downloaded the file to some other location.
+
+.. code-block:: psql
+
+  \copy companies
+    from 'companies.csv' with csv;
+  \copy campaigns
+    from 'campaigns.csv' with csv;
+  \copy ads (id, company_id, campaign_id, name, image_url, target_url,
+             impressions_count, clicks_count, created_at, updated_at)
+    from 'ads.csv' with csv;
+  \copy clicks
+    from 'clicks.csv' with csv;
+  \copy impressions
+    from 'impressions.csv' with csv;
+
+Querying the Cluster
```

Comment:
I have two high level comments on this section.

(1) We should communicate the benefits of running these queries to app devs too i.e. what is the implication that all constructs work.

i.e. if you model your data in the above way, very minimal changes are needed to your application to scale as things work very easily. Citus will support all construct which your ORM generates. Then, we can also inform users about the MT gem which applies to ruby devs.

(2) Also, maybe add a concluding paragraph driving home the point that anything including inserts / updates / deletes, complex SQL, transactions, will work if you are scoped to a tenant. (As a takeway). We could then (maybe?) add the content I mentioned above.

What do you think?